pandas/core/strings.py

```diff
@@ -649,11 +650,14 @@ def str_split(arr, pat=None, n=None, return_type='series'):
     from pandas.core.frame import DataFrame
     from pandas.core.index import Index
 
```

Comment:
this is a minor point but it might be good idea to use a `dict` to organize the mapping from old to new arg names, at the beginning of the function, something like:

```
arg_map = {'series': 'same', 'index': 'same', 'frame': 'expand'}
if return_type in arg_map:
    warnings.warn("return_type='%s' is deprecated, please use '%s' instead" % (return_type, arg_map[return_type]), FutureWarning)
    return_type = arg_map[return_type]
```

this way the rest of the code doesn't have to contain references to the old names, making it easier to remove the deprecation stuff later.